### PR TITLE
[close #123] Update tikv-br readme doc

### LINKED ISSUE #123: Update TiKV-BR README.md
### 1. Describe the bug

README.md in migration/br is outdated. Please update it.

### 5. What is your migration tool and TiKV version? (Required)

- TiKV:
- TiKV CDC:
- TiKV BR:
- TiKV Online Bulk Load:

### PR DESCRIPTION
**Pull request by haojinming**
Signed-off-by: haojinming <jinming.hao@pingcap.com>

### What problem does this PR solve?

Issue Number: close #123 

Problem Description: 
TiKV-BR readme doc is outdated.

### What is changed and how does it work?
Update tikv-br readme doc

### Code changes

- No code

### Check List for Tests

This PR has been tested by at least one of the following methods:
- No code

### Side effects

- No side effects

### Related changes

- No related changes

### CODE REVIEW

File: br/README.md [hunk 1]
```diff
@@ -1,131 +1,78 @@
...
+本文介绍了 TiKV-BR 的工作原理、推荐部署配置、使用限制以及几种使用方式。
 
-**Backup & Restore (BR)** is a command-line tool for distributed backup and restoration of the TiDB cluster data.
+## 工作原理
 
-## Architecture
+TiKV-BR 将备份或恢复操作命令下发到各个 TiKV 节点。TiKV 收到命令后执行相应的备份或恢复操作。
+在一次备份或恢复中，各个 TiKV 节点都会有一个对应的备份路径，TiKV 备份时产生的备份文件将会保存在该路径下，恢复时也会从该路径读取相应的备份文件。
 
 <img src="docs/images/arch.svg?sanitize=true" alt="architecture" width="600"/>
```
Comment (haojinming) [inline br/README.md:11]: update the picture.

File: br/README.md [hunk 2]
```diff
@@ -1,131 +1,79 @@
-# BR
+# TiKV-BR
```
Comment (pingyu) [inline br/README.md:1]: Suggest to keep the English readme and add a new `README-cn.md` for Chinese.

File: br/README.md [hunk 3]
```diff
@@ -1,131 +1,79 @@
...
 
-**Backup & Restore (BR)** is a command-line tool for distributed backup and restoration of the TiDB cluster data.
+## 工作原理
 
-## Architecture
+TiKV-BR 将备份或恢复操作命令下发到各个 TiKV 节点。TiKV 收到命令后执行相应的备份或恢复操作。
+在一次备份或恢复中，各个 TiKV 节点都会有一个对应的备份路径，TiKV 备份时产生的备份文件将会保存在该路径下，恢复时也会从该路径读取相应的备份文件。
 
-<img src="docs/images/arch.svg?sanitize=true" alt="architecture" width="600"/>
+<img src="docs/images/tikv-architecture.png?sanitize=true" alt="architecture" width="600"/>
```
Comment (pingyu) [inline br/README.md:11]: Wrong image filename. It is `tikv-br architecture.png`.

File: br/README.md [hunk 4]
```diff
@@ -1,131 +1,79 @@
...
 
-## Documentation
+## 备份文件类型
 
-[Chinese Document](https://docs.pingcap.com/zh/tidb/v4.0/backup-and-restore-tool)
+备份路径下会生成以下两种类型文件：
+- SST 文件：存储 TiKV 备份下来的数据信息, SST 文件的数据格式根据 `backup` 时指定的 `dst-api-version` 来决定
```
Comment (pingyu) [inline br/README.md:16]:
```suggestion
- SST 文件：存储 TiKV 备份下来的数据信息，SST 文件的数据格式根据 `backup` 时指定的 `dst-api-version` 来决定
```

File: br/README.md [hunk 5]
```diff
@@ -1,131 +1,79 @@
...
-mysql -uroot --host 127.0.0.1 -P4000 -E -e "USE test; DROP TABLE order_line; show tables" -u root -p
+### 最佳实践
+下面是使用 TiKV-BR 进行备份恢复的几种推荐操作：
+- 推荐在业务低峰时执行备份操作，这样能最大程度地减少对业务的影响。
+- TiKV-BR 支持在不同拓扑的集群上执行恢复，但恢复期间对在线业务影响很大，建议低峰期或者限速 (rate-limit) 执行恢复。
+- TiKV-BR 备份最好串行执行。不同备份任务并行会导致备份性能降低，同时也会影响在线业务。
+- TiKV-BR 恢复最好串行执行。不同恢复任务并行会导致 Region 冲突增多，恢复的性能降低。
+- 推荐在 -s 指定的备份路径上挂载一个共享存储，例如 NFS。这样能方便收集和管理备份文件。
+- 在使用共享存储时，推荐使用高吞吐的存储硬件，因为存储的吞吐会限制备份或恢复的速度。
+- TiKV-BR 默认会分别在备份、恢复完成后，可以通过指定 --checksum=true进行一轮数据校验，将文本数据同集群数据比较，来保证正确性。如果 TiKV 集群开启了 TTL，请注意，在备份、恢复过程中如果有数据 TTL 失效，引发数据过期删除，此时 checksum 会不一致，此时并不代表数据存在一致性问题。
```
Comment (pingyu) [inline br/README.md:44]:
```suggestion
- 可以通过指定 `--checksum=true`，在备份、恢复完成后进行一轮数据校验。数据校验将分别计算备份数据与 TiKV 集群中数据的 checksum，并对比二者是否相同。请注意，如果需要进行数据校验，请确保在备份或恢复的全过程，TiKV 集群没有数据变更和 TTL 过期。
```

建议 checksum 默认关闭，只有在用户明确没有数据变更以及潜在的 TTL 过期时，才打开 checksum

File: br/README.md [hunk 6]
```diff
@@ -1,131 +1,79 @@
...
+- TiKV-BR 支持不同 api-version 的升级备份，即可以通过指定 `--dst-api-version V2` 将 `api-version=1` 的 TiKV 集群备份为 `api-version=2` 的 `SST` 文件，并将备份文件恢复到 `api-version=2` 的 TiKV 集群中。
```
Comment (pingyu) [inline br/README.md:45]:
```suggestion
- TiKV-BR 可用于实现 [`api-version`](https://docs.pingcap.com/zh/tidb/stable/tikv-configuration-file#api-version-%E4%BB%8E-v610-%E7%89%88%E6%9C%AC%E5%BC%80%E5%A7%8B%E5%BC%95%E5%85%A5) 从 V1 到 V2 的集群数据迁移。通过指定 `--dst-api-version V2` 将 `api-version=1` 的 TiKV 集群备份为 V2 格式，然后将备份文件恢复到新的 `api-version=2` TiKV 集群中。
```

File: br/README.md [hunk 7]
```diff
@@ -1,131 +1,79 @@
...
+tikv-br backup raw --pd "&{PDIP}:2379" -s "local:///tmp/backup" --dst-api-version v2 --log-file="/tmp/br_backup.log
+```
+命令行各部分的解释如下：
+- `backup`：`tikv-br` 的子命令
+- `raw`：`backup` 的子命令
+- `-s` 或 `--storage`：备份保存的路径
+- `"local:///tmp/backup"`：`-s` 的参数，保存的路径为各个 TiKV 节点本地磁盘的 `/tmp/backup`
+- `--pd`：`PD` 服务地址
+- `"${PDIP}:2379"`：`--pd` 的参数
+- `--dst-api-version`: 指定备份文件的 `api-version`，请见 tikv config
```
Comment (pingyu) [inline br/README.md:62]:
```suggestion
- `--dst-api-version`: 指定备份文件的 `api-version`，请见 [tikv-server config](https://docs.pingcap.com/zh/tidb/stable/tikv-configuration-file#api-version-%E4%BB%8E-v610-%E7%89%88%E6%9C%AC%E5%BC%80%E5%A7%8B%E5%BC%95%E5%85%A5)
```

File: br/README.md [hunk 8]
```diff
@@ -1,131 +1,79 @@
...
+- `v2`: `--dst-api-version` 的参数，可选参数为 `v1`, `v1ttl`, `v2`(不区分大小写), 如果不指定 `dst-api-version` 参数，则备份文件的 `api-version` 与指定 `--pd` 所属的 TiKV 集群 `api-version` 相同。
```
Comment (pingyu) [inline br/README.md:63]: 仅支持 `v2`  ?

File: br/README.md [hunk 9]
```diff
@@ -1,131 +1,79 @@
...
+要将 Raw 模式备份数据恢复到集群中来，可使用 `tikv-br restore raw` 命令。该命令的使用帮助可以通过 `tikv-br restore raw --help` 来获取。
+用例：将 `/tmp/backup` 路径中的 Raw 模式备份数据恢复到集群中。
+```
+tikv-br restore raw \
+    --pd "${PDIP}:2379" \
+    --storage "local:///tmp/backup" \
+    --log-file restoreraw.log
+```
+以上命令中，`--log-file` 选项指定把 `TiKV-BR` 的 log 写到 `restoreraw.log` 文件中。
+恢复期间还有进度条会在终端中显示，当进度条前进到 100% 时，说明恢复已完成。可以通过指定 `--checksum=true` 在 `restore` 结束时进行一轮数据校验，将文本数据同集群数据比较，来保证正确性。请注意: 如果 TiKV 集群开启了 `TTL`，在备份恢复过程中如果有数据 TTL 失效，引发数据过期删除，此时 `checksum` 不一致，并不代表数据存在一致性问题。
```
Comment (pingyu) [inline br/README.md:76]: 同上。请确保没有数据变更与 TTL 失效。

File: br/README.md [hunk 10]
```diff
@@ -1,131 +1,79 @@
...
+一条 `tikv-br` 命令是由子命令、选项和参数组成的。子命令即不带 `-` 或者 `--` 的字符。选项即以 `-` 或者 `--` 开头的字符。参数即子命令或选项字符后紧跟的、并传递给命令和选项的字符。
+#### 备份集群 Raw 模式数据
+要备份 TiKV 集群中 Raw 模式数据，可使用 `tikv-br backup raw` 命令。该命令的使用帮助可以通过 `tikv-br backup raw --help` 来获取。
+用例：将 TiKV 集群中 Raw 模式数据备份到 `/tmp/backup` 目录中。
 ```
-
-## Compatibility test
-
-See [COMPATBILE_TEST](./COMPATIBLE_TEST.md)
+tikv-br backup raw --pd "&{PDIP}:2379" -s "local:///tmp/backup" --dst-api-version v2 --log-file="/tmp/br_backup.log
```
Comment (pingyu) [inline br/README.md:53]: 建议加上 checksum。同恢复的例子。

File: br/README.md (hunk 8 above)
Comment (haojinming) [reply br/README.md:63]: 从 apiv1 的集群执行 backup 的话可以用 v1 或 v1ttl 的

Comment (pingyu) [reply br/README.md:63]: 我是觉得那样简单。v1 <-> v1ttl 好像也没这个需求场景
支持 v1 和 v1ttl 也行吧，但就需要备注一下不支持 v2 -> v1/v1ttl ？

File: br/README.md (hunk 10 above)
Comment (haojinming) [reply br/README.md]: done

File: br/README.md (hunk 9 above)
Comment (haojinming) [reply br/README.md]: done

File: br/README.md (hunk 5 above)
Comment (haojinming) [reply br/README.md]: done, checksum 现在默认是  false 的

**Comment by haojinming** [comment]
Review comments are addressed. PTAL~ @pingyu @zeminzhou

File: br/README-cn.md [hunk 11]
```diff
@@ -0,0 +1,86 @@
+# TiKV-BR
+
+**TiKV Backup & Restore (TiKV-BR)** 是 TiKV 分布式备份恢复的命令行工具, 用于对 TiKV 集群进行数据备份和恢复。
```
Comment (pingyu) [inline br/README-cn.md:3]:
```suggestion
**TiKV Backup & Restore (TiKV-BR)** 是 TiKV 分布式备份恢复的命令行工具，用于对 TiKV 集群进行数据备份和恢复。
```
建议把英文逗号都替换为中文逗号吧。这样统一一点。

File: br/README-cn.md [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+tikv-br backup raw --pd "&{PDIP}:2379" -s "local:///tmp/backup" --dst-api-version v2 --log-file="/tmp/br_backup.log
+```
+命令行各部分的解释如下：
+- `backup`：`tikv-br` 的子命令
+- `raw`：`backup` 的子命令
+- `-s` 或 `--storage`：备份保存的路径
+- `"local:///tmp/backup"`：`-s` 的参数, 保存的路径为各个 TiKV 节点本地磁盘的 `/tmp/backup`
+- `--pd`：`PD` 服务地址
+- `"${PDIP}:2379"`：`--pd` 的参数
+- `--dst-api-version`: 指定备份文件的 `api-version`，请见 [tikv-server config](https://docs.pingcap.com/zh/tidb/stable/tikv-configuration-file#api-version-%E4%BB%8E-v610-%E7%89%88%E6%9C%AC%E5%BC%80%E5%A7%8B%E5%BC%95%E5%85%A5)- `v2`: `--dst-api-version` 的参数, 可选参数为 `v1`, `v1ttl`, `v2`(不区分大小写), 如果不指定 `dst-api-version` 参数, 则备份文件的 `api-version` 与指定 `--pd` 所属的 TiKV 集群 `api-version` 相同。
```
Comment (pingyu) [inline br/README-cn.md:62]:
```suggestion
- `--dst-api-version`: 指定备份文件的 `api-version`，请见 [tikv-server config](https://docs.pingcap.com/zh/tidb/stable/tikv-configuration-file#api-version-%E4%BB%8E-v610-%E7%89%88%E6%9C%AC%E5%BC%80%E5%A7%8B%E5%BC%95%E5%85%A5)
- `v2`: `--dst-api-version` 的参数, 可选参数为 `v1`, `v1ttl`, `v2`(不区分大小写), 如果不指定 `dst-api-version` 参数, 则备份文件的 `api-version` 与指定 `--pd` 所属的 TiKV 集群 `api-version` 相同。  
```

**Review by pingyu — COMMENTED** [review]
Rest LGTM~

**Review by zeminzhou — APPROVED** [review]

**Review by pingyu — APPROVED** [review]
LGTM~

**Pull request merged**